packages/unchained-client/src/evm/ethereum/parser/yearn.ts

```diff
@@ -61,9 +64,18 @@ export class Parser implements SubParser<Tx> {
     // failed to decode input data
     if (!decoded) return
 
+    let approveValue: string | null = null
+    let assetId: string | null = null
+
     switch (txSigHash) {
       case this.supportedYearnFunctions.approveSigHash:
         if (decoded?.args._spender !== SHAPE_SHIFT_ROUTER_CONTRACT) return
+        approveValue = decoded?.args._value?.toString()
+        assetId = toAssetId({
+          ...fromChainId(this.chainId),
+          assetNamespace: 'erc20',
+          assetReference: tx.to
+        })
```

Comment:
Not entirely sure about this one either, but IMO, better to only set `assetId` for `approve()`s.
Setting it for other calls could produce unwanted results like having the contract we're interacting to as an `AssetId` e.g ShapeShift DAO router) and would be a pain to actually parse what's an actual token contract being interacted with as an `approve()` and what's not in web